cmake/CheckLibc.cmake

```diff
@@ -29,3 +29,23 @@ if (Strerror_R_Returns_Char_P)
     PRIVATE
       SEASTAR_STRERROR_R_CHAR_P)
 endif ()
+
+check_cxx_source_compiles ("
```

Comment:
instead of using an application, we could just use
```cmake
include (CheckFunctionExists)

check_function_exists (pthread_attr_setaffinity_np
  Pthread_Attr_Setaffinity_Np)
```

simpler this way.

Reply:
okay changed it